eda/src/autogluon/eda/analysis/interaction.py

```diff
@@ -210,3 +208,242 @@ def _fit(self, state: AnalysisState, args: AnalysisState, **fit_kwargs):
+        "powerlaw",
+        "powerlognorm",
+        "powernorm",
+        "rdist",
+        "reciprocal",
+        "rayleigh",
+        "rice",
+        "recipinvgauss",
+        "semicircular",
+        "t",
+        "triang",
+        "truncexpon",
+        "truncnorm",
+        "tukeylambda",
+        "uniform",
+        "vonmises",
+        "vonmises_line",
+        "wald",
+        "weibull_min",
+        "weibull_max",
+    ]
+
+    def __init__(
+        self,
+        columns: Union[str, List[str]],
+        pvalue_min: float = 0.01,
+        keep_top_n: Optional[int] = None,
+        distributions_to_fit: Optional[Union[str, List[str]]] = None,
+        parent: Optional[AbstractAnalysis] = None,
+        children: Optional[List[AbstractAnalysis]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(parent, children, **kwargs)
+
+        if keep_top_n is None and distributions_to_fit is None:
+            keep_top_n = 3
+
+        if isinstance(columns, str):
+            columns = [columns]
+        self.columns = columns
+
+        self.pvalue_min = pvalue_min
+        self.keep_top_n = keep_top_n
+
+        if distributions_to_fit is None:
+            distributions_to_fit = self.AVAILABLE_DISTRIBUTIONS
+        if isinstance(distributions_to_fit, str):
+            distributions_to_fit = [distributions_to_fit]
+        not_supported = [d for d in distributions_to_fit if d not in self.AVAILABLE_DISTRIBUTIONS]
+        if len(not_supported) > 0:
+            raise ValueError(
+                f"The following distributions are not supported: {sorted(not_supported)}. "
+                f"Supported distributions are {sorted(self.AVAILABLE_DISTRIBUTIONS)}"
+            )
+        self.distributions_to_fit = distributions_to_fit
+
+    def can_handle(self, state: AnalysisState, args: AnalysisState) -> bool:
+        return True
+
+    def _fit(self, state: AnalysisState, args: AnalysisState, **fit_kwargs) -> None:
+        state.distributions_fit = {}
+        for (ds, df) in self.available_datasets(args):
+            state.distributions_fit[ds] = {}
+            for c in self.columns:
+                if c in df.columns:
+                    col = df[c]
+                    col = col[col.notna()]  # skip NaNs
+                    dist = self._fit_dist(col, self.pvalue_min)
+                    if dist is not None:
+                        state.distributions_fit[ds][c] = dist
+
+    def _fit_dist(self, series, pvalue_min=0.01):
+        results = {}
+        if not is_numeric_dtype(series):
+            self.logger.warning(f"{series.name}: distribution cannot be fit; only numeric columns are supported")
+            return None
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            for i in self.distributions_to_fit:
+                dist = getattr(stats, i)
+                param = dist.fit(series)
+                statistic, pvalue = stats.kstest(series, i, args=param)
```

Comment:
I assume this will only work for continuous dist so worth to check if only continuous are supported

Reply:
We check if the type is numeric; categorical types are not supported.